src/rnanorm/annotation.py

```diff
@@ -0,0 +1,70 @@
+"""Annotation parsing methods."""
+
+import gzip
+import re
+from collections import defaultdict
+
+import pandas as pd
+
+
+def universal_open(file_name, *args, **kwargs):
+    """Use correct open function, based on filename extension."""
+    if file_name.endswith(".gz"):
+        return gzip.open(file_name, *args, **kwargs)
+    return open(file_name)
+
+
+def get_gene_length(starts_ends):
+    """Compute gene length as a sum of it's exons lengths."""
+    # Merge all exons of a single gene. One could use pybedtools.merge(),
+    # but simple hand-built merge works much-much faster!
+    # From here: https://stackoverflow.com/questions/43600878/merging-overlapping-intervals
+    starts_ends = sorted(starts_ends, key=lambda interval: interval[0])
+    merged = [starts_ends[0]]
+    for current in starts_ends:
+        previous = merged[-1]
+        if current[0] <= previous[1]:
+            previous[1] = max(previous[1], current[1])
+        else:
+            merged.append(current)
+
+    # Compute gene length from exon sizes
+    return sum(end - start for (start, end) in merged)
+
+
+def get_gene_lengths(gtf):
+    """Compute gene lengths from GTF file."""
+    # 1. Group exon start/end coordinates by gene ID
+    # One could use pybedtools, but hand-built parsing is ~10x faster
+    data = defaultdict(lambda: defaultdict(list))
+    with universal_open(gtf, "rt") as handle:
+        for row in handle:
+            row = row.split("\t")
```

Comment:
Let's avoid parsing the annotation file manually. Use pybedtools or at least the csv reader.